custom_components/waste_collection_schedule/waste_collection_schedule/source/oslokommune_no.py

```diff
@@ -30,14 +37,38 @@
     "":           "mdi:trash-can",
     "restavfall": "mdi:trash-can",
     "papir":      "mdi:newspaper-variant-multiple"
-} 
+}
+
+# ### Arguments affecting the configuration GUI ####
+
+PARAM_DESCRIPTIONS = {  # Optional dict to describe the arguments, will be shown in the GUI configuration below the respective input field
+    "en": {
+        "point_id": "Optional waste point ID to filter by",
+    },
+}
+
+PARAM_TRANSLATIONS = {  # Optional dict to translate the arguments, will be shown in the GUI configuration form as placeholder text
+    "en": {
+        "point_id": "Point ID",
+    },
+}
+
+# ### End of arguments affecting the configuration GUI ####
 
 class Source:
-    def __init__(self, street_name, house_number, house_letter, street_id):
-        self._street_name  = street_name
+    def __init__(
+        self,
+        street_name,
+        house_number,
+        street_id,
+        house_letter = None,
+        point_id = None,
```

Comment:
```suggestion
        point_id:int|None = None,
```

Adding this type hint will enforce an integer when configuring through the HA GUI


If you're at it, you could type hint the other ones as well (not relevant for config as it assumes str if no type hint given.

Reply:
Yep, I think I tried this first, but then reverted it when I saw HA's form input for int was the kind with inline arrow buttons, and it seemed wrong for an ID input. Might be some way to avoid that?

Reply:
We use the default integer input from HA, You can modify it quite a bit by using the Integer Selector (https://www.home-assistant.io/docs/blueprint/selectors/#number-selector)

But I don't think that's really necessary.

I personally think the arrows are fine as you can input with your keyboard OR use the arrows and the errors automatically indicate to the user to input an integer

Reply:
Fixed!